apps/src/dance/lab2/views/DanceView.tsx

```diff
@@ -489,12 +511,32 @@ const DanceView: React.FunctionComponent<{
     usingMusicProject,
   ]);
 
+  // Create dance validator.
+  const danceValidator = useMemo(
+    () =>
+      new DanceValidator(() => {
+        return currentCondition.current;
+      }),
+    [currentCondition]
+  );
+
+  // Pass dance validator to Progress Manager.
+  useEffect(() => {
+    if (
+      guideMode === 'instructions' &&
+      progressManager &&
+      levelProperties.appName === 'dance'
```

Comment:
Technically speaking this condition should be redundant as we should never mount DanceView if the app type isn't dance